### PR TITLE
Samson192 (adding additional status to the show deploys screen)

### PR DESCRIPTION
**Pull request by irwaters**
Adding some additional information to the display deploy screen.  Includes job_id, pid of the shell running the command and how many seconds it's been running

screen shot 2016-03-17 at 5 03 18 pm
/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link:
https://zendesk.atlassian.net/projects/SAMSON/issues/SAMSON-192

### Risks
- Level: Low/Med/High
Low

### CODE REVIEW

**Comment by zendesknotifications** [comment]
&nbsp;<b>Code Climate</b> has <a href="https://codeclimate.com/repos/53340bef6956800b9000675c/pull/764">analyzed this pull request</a>.

File: app/helpers/status_helper.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def status_label(key)
 
   def status_panel(deploy)
     content = h deploy.summary
+    if deploy.active?
+      content << content_tag(:ul)
```
Comment (grosser) [inline app/helpers/status_helper.rb]: a lone ul and then text ... how about:

`content_tag(:ul, content_tag(:li, deploy.summary_for_process))` ?

File: app/models/deploy.rb [hunk 2]
```diff
@@ -26,6 +26,11 @@ def summary
     "#{job.user.name} #{deploy_buddy} #{summary_action} #{short_reference} to #{stage.name}"
   end
 
+  def summary_for_process
+		t = (Time.now.utc.to_i - start_time.utc.to_i)
```
Comment (grosser) [inline app/models/deploy.rb]: strange witespace

File: app/models/job.rb [hunk 3]
```diff
@@ -108,6 +108,11 @@ def full_url
     deploy.try(:full_url) || AppRoutes.url_helpers.project_job_url(project, self)
   end
 
+  def pid
+    return execution.pid if execution
```
Comment (grosser) [inline app/models/job.rb]: -> `execution.try :pid` (no return and no nil)

File: test/controllers/streams_controller_test.rb [hunk 4]
```diff
@@ -18,6 +18,9 @@
         fake_execution = JobExecution.new("foo", job)
         JobExecution.expects(:find_by_id).returns(fake_execution)
 
+        # make sure that the JobExecution object responds to the pid method
+        assert fake_execution.respond_to?(:pid)
```
Comment (grosser) [inline test/controllers/streams_controller_test.rb]: not the place to test this ... test in the job_test 

```
it 'has a pid when the execution is running'
it 'has no pid when the execution is finished'
```

**Comment by grosser** [comment]
Please use descriptive titles for PRs

Do we need the job-id ? So far I only saw people wanting to kill jobs, so the pid is needed ...

**Comment by irwaters** [comment]
tests are in/pssing

File: app/models/deploy.rb [hunk 5]
```diff
@@ -26,6 +26,11 @@ def summary
     "#{job.user.name} #{deploy_buddy} #{summary_action} #{short_reference} to #{stage.name}"
   end
 
+  def summary_for_process
+    t = (Time.now.utc.to_i - start_time.utc.to_i)
```
Comment (grosser) [inline app/models/deploy.rb]: everything is utc on our servers

File: test/models/job_test.rb [hunk 6]
```diff
@@ -1,6 +1,15 @@
 require_relative '../test_helper'
+require 'byebug'
```
Comment (grosser) [inline test/models/job_test.rb]: that can go away now ?

File: test/models/job_test.rb [hunk 7]
```diff
@@ -10,4 +19,20 @@
       Job.create.errors[:project].must_equal(['is locked'])
     end
   end
+
+  describe "looking for the pid of a job" do
+    before do
+      JobExecution.start_job(JobExecution.new('master', job))
+      JobExecution.any_instance.stubs(:pid).returns(1234)
+    end
+    it "has a pid when running" do
```
Comment (grosser) [inline test/models/job_test.rb]: newline before each `it` plz

File: test/models/job_test.rb [hunk 8]
```diff
@@ -10,4 +19,20 @@
...
+      job.run!
+      job.pid.wont_be_nil
+    end
+    it "has no pid when stopped" do
+      job.stop!
+      job.pid.must_be_nil
+    end
+  end
+
```
Comment (grosser) [inline test/models/job_test.rb]: and no empty newlines :)

File: test/models/job_test.rb [hunk 9]
```diff
@@ -10,4 +19,20 @@
       Job.create.errors[:project].must_equal(['is locked'])
     end
   end
+
+  describe "looking for the pid of a job" do
```
Comment (grosser) [inline test/models/job_test.rb]: we are on job, so a `describe "#pid"` is enough :)

File: test/controllers/streams_controller_test.rb (hunk 4 above)
Comment (grosser) [inline test/controllers/streams_controller_test.rb]: this can go away now ?

**Comment by grosser** [comment]
:+1:

**Comment by grosser** [comment]
screen shot 2016-03-22 at 10 15 12 am

@irwaters  seems to not work for deploys that are pending approval ...

**Comment by grosser** [comment]
nvm ... this was actually running for 100 hours ...

**Comment by grosser** [comment]
seems to break macro execution in production:

```
ActionView::Template::Error (undefined method `summary_for_process' for #<Job:0x007fa2d018c0e8>):
    1: <%= status_panel @job %>
  app/helpers/status_helper.rb:25:in `status_panel'
  app/views/jobs/_header.html.erb:1:in `_app_views_jobs__header_html_erb__4278462716950424045_70167894051740'
  app/views/jobs/show.html.erb:15:in `_app_views_jobs_show_html_erb___1361811445455127576_70167970172060'
  app/controllers/jobs_controller.rb:35:in `show'
```

@irwaters please add a test that shows this error and then fix it :)

/cc @lukkry

**Pull request merged**